PowerValidatedSolutions.psm1

```diff
@@ -20714,6 +20721,384 @@ Function Get-PasswordPolicyConfig {
+			    			Write-LogMessage -Type ERROR -Message "$($product.Name):$($section.Name):$($parameterName.Name) parameter is not configured, please check the custom password policy json file" -Colour Red
+			    			$encounterError = "True"
+			    			break
+			    		}
+			    	}
+			    }
+            }
+        }
+    }
+
+    # Validating parameter values
+    if ($encounterError -ne "True") {
+        foreach ($product in $customProductList) {
+            $customSectionList = $customConfig.($product.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+            foreach ($section in $customSectionList) {
+                $defaultParameterList = $defaultConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+                $customParameterList = $customConfig.($product.Name).($section.Name) | Get-Member | where {$_.MemberType -match "NoteProperty"} | Select-Object Name
+                # Write-LogMessage -Type INFO -Message "Validating $($product.Name):$($section.Name) parameters value range"
+                foreach ( $parameterName in $customParameterList) {
+                    # Validating parameter values
+                    switch ($parameterName.Name)  {
+                        # Password Expiration Section
+                        "maxDays"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):maxDays" -value $customConfig.($product.Name).($section.Name)."maxDays" -minRange 1 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "minDays"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):minDays" -value $customConfig.($product.Name).($section.Name)."minDays" -minRange 0 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "warningDays"
+                        {
+                            $checkReturn = checkRange -name "$($product.Name):$($section.Name):warningDays" -value $customConfig.($product.Name).($section.Name)."warningDays" -minRange 0 -maxRange 99999 -required $true
+                            if (-Not $checkReturn) {
+                                $encounterError = "True"
+                            }
+                        }
+                        "email"
+                        {
+                            if ($customConfig.($product.Name).($section.Name)."email" -eq "") {
+                                Write-LogMessage -Type WARNING -Message "$($product.Name):$($section.Name):email is not configured."  -Colour Yellow
```

Comment:
Change to:

Write-LogMessage -Type ERROR -Message "Parameter ($($product.Name):$($section.Name):email) Not Configured, Please Check the Password Policy Configuration File and Run Again" -Colour Red

Reply:
resolved in 2948bc273bae34312514db8aba5d7cb584efd8d4